### ISSUE TITLE
Incorrect crop area output when the input is SVG

### ISSUE THREAD

**Issue by JohnCido** [ValentinH/react-easy-crop#381]
**Describe the bug**
When trying to crop an SVG file, the final output cropped area is wrong. So any code that relies on this output will generate wrong image output.

**To Reproduce**
1. Open [this sandbox](https://codesandbox.io/s/gj6kk)
2. Download this image
![crop test file](https://user-images.githubusercontent.com/7105485/167259337-7fdb4b99-6631-46c2-834e-8f9ca721f00b.svg)
3. Open this file in the sandbox
4. Crop it and see the output

**Expected behavior**
Crop the exact area that the user has defined using the cropper.

- Please create a CodeSandbox that demonstrates your issue. You can fork the Basic example: https://codesandbox.io/s/react-easy-crop-v69ly910ql.

**Comment by ValentinH** [comment]
Indeed, something is wrong with the sizes computation. First thing I noticed is that `naturalWidth`/`naturalHeight` doesn't exist for SVGs and we are using this to measure to media size here:
https://github.com/ValentinH/react-easy-crop/blob/4e31ef20a95f3e759aedc57e70788d778067eb52/src/Cropper.tsx#L245-L246

I guess we should use `width/height` in that case but we need to check if something else needs to be adjusted.

**Comment by raed667** [comment]
Cropping works with some other SVGs, for example this image https://user-images.githubusercontent.com/1442690/182855138-e84d1916-8a19-4152-89a8-4c9074f54dfd.svg

Could it be that the difference is that the original svg has `width="100%" height="100%" viewBox="0 0 3200 3200" ` and my example has `width="640" height="480" viewBox="0 0 640 480"` ?

**Comment by tonyfarney** [comment]
I am facing the same issue here. Is there some workaround?

**Comment by 0xturner** [comment]
I've encountered the same issue. Only SVG's with a fixed height and width are cropped correctly.

Not a perfect solution, but a workaround that worked for my use case is using the view box dimensions as a fallback values if a fixed height and width are not present. I preprocess SVG files, adding a fixed height and width, before passing to `react-easy-crop`'s `Cropper`

```typescript
const svg = svgDoc.querySelector("svg");

if (
  !svg ||
  (isFixedPixelValue(svg.getAttribute("width")) &&
    isFixedPixelValue(svg.getAttribute("height")))
) {
  return;
}

const [, , viewBoxWidth, viewBoxHeight] =
  svg.getAttribute("viewBox")?.split(/[\s,]+/) ?? [];

if (!viewBoxWidth || !viewBoxHeight) {
  return;
}

svg.setAttribute("width", viewBoxWidth);
svg.setAttribute("height", viewBoxHeight);

...

function isFixedPixelValue(value: string | null) {
  return value && (/^\d*\.?\d+px$/.test(value) || /^\d*\.?\d+$/.test(value));
}
```
